ledger/bulletin.go

```diff
@@ -83,11 +84,29 @@ func (b *bulletin) Wait(round basics.Round) chan struct{} {
 		signal = makeNotifier()
 		b.pendingNotificationRequests[round] = signal
 	}
+	// Increment count of waiters, to support canceling.
+	signal.count++
+
 	return signal.signal
 }
 
+// CancelWait removes a wait for a particular round. If no one else is waiting, the
+// notifier channel for that round is removed.
+func (b *bulletin) CancelWait(round basics.Round) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	signal, exists := b.pendingNotificationRequests[round]
+	if exists {
+		signal.count--
+		if signal.count <= 0 {
+			delete(b.pendingNotificationRequests, round)
+		}
+	}
+}
```

Comment:
Don't you need to write the new lower `signal.count` back into the map, if it didn't go to 0?

Reply:
Nevermind, that's why you switched to pointers, eh?